File: tests/journal.lib/file_sanity.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <p2/journal.h>
// support
#include <cassert>
#include <fstream>
#include <filesystem>


// alias the type
using stream_t = pyre::journal::stream_t;


// exercise the stream device
int main() {
    // the path of the file
    auto filename = std::filesystem::path("file_sanity.out");

    // make a file stream
    auto ofs = std::ofstream(filename);
    // instantiate
    stream_t stream(filename, ofs);
    // check its name
    assert (stream.name() == filename);

    // clean up
    // close the file
    ofs.close();

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/file_sanity.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


def test():
    """
    Verify that we can build a stream based device
    """
    # pick a file name
    filename = "file_sanity.log"

    # get the device
    from j2.File import File
    # instantiate
    device = File(path=filename)

    # check its name
    assert device.name == "log"
    # and the path
    assert device.path == filename

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
